projects/packages/forms/src/blocks/contact-form/components/jetpack-field-number.js

```diff
@@ -0,0 +1,153 @@
+import { useBlockProps } from '@wordpress/block-editor';
+import { createBlock, getDefaultBlockName } from '@wordpress/blocks';
+import { __experimentalNumberControl as NumberControl } from '@wordpress/components'; // eslint-disable-line @wordpress/no-unsafe-wp-apis
+import { createHigherOrderComponent, compose } from '@wordpress/compose';
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
+import { isEmpty } from 'lodash';
+import { useFormStyle } from '../util/form';
+import { withSharedFieldAttributes } from '../util/with-shared-field-attributes';
+import JetpackFieldControls from './jetpack-field-controls';
+import JetpackFieldLabel from './jetpack-field-label';
+import { useJetpackFieldStyles } from './use-jetpack-field-styles';
+
+const JetpackNumberField = props => {
+	const {
+		attributes,
+		clientId,
+		id,
+		isSelected,
+		required,
+		requiredText,
+		label,
+		setAttributes,
+		placeholder,
+		min,
+		max,
+		width,
+		insertBlocksAfter,
+	} = props;
+
+	const { blockStyle, fieldStyle } = useJetpackFieldStyles( attributes );
+	const formStyle = useFormStyle( clientId );
+	const blockProps = useBlockProps( {
+		className: clsx( 'jetpack-field', {
+			'is-selected': isSelected,
+			'has-placeholder': ! isEmpty( placeholder ),
+		} ),
+		style: blockStyle,
+	} );
+
+	return (
+		<>
+			<div { ...blockProps }>
+				<JetpackFieldLabel
+					attributes={ attributes }
+					label={ label }
+					required={ required }
+					requiredText={ requiredText }
+					setAttributes={ setAttributes }
+					style={ formStyle }
+				/>
+				<input
+					className="jetpack-field__input"
+					onChange={ e => setAttributes( { placeholder: e.target.value } ) }
+					style={ fieldStyle }
+					type="number"
+					value={ placeholder }
+					min={ min }
+					max={ max }
+					onKeyDown={ event => {
+						if ( event.defaultPrevented || event.key !== 'Enter' ) {
+							return;
+						}
+						insertBlocksAfter( createBlock( getDefaultBlockName() ) );
+					} }
+				/>
+			</div>
+
+			<JetpackFieldControls
+				id={ id }
+				required={ required }
+				width={ width }
+				setAttributes={ setAttributes }
+				placeholder={ placeholder }
+				attributes={ attributes }
+				hidePlaceholder={ true }
+				extraFieldSettings={ [
```

Comment:
Quite dislike adding this whole file just to add two fields under `extraFieldSettings`.

Could potentially be done differently, directly in `jetpack-field.js` with conditionally by importing the settings only from another file. 🤔

Reply:
I did a version of this with `extraFieldSettings` and `min/max` passed as props to `jetpackField` but it also didn't feel great so let's go with this separate files approach for now. When untangling these to child blocks I think we'll get to more sensible way to do abstractions.